src/MySQLReplication/JsonBinaryDecoder/JsonBinaryDecoderService.php

```diff
@@ -238,11 +243,21 @@ private static function valueEntrySize(bool $large): int
      * @throws BinaryDataReaderException
      * @throws JsonBinaryDecoderException
      */
-    private function getOffsetOrInLinedValue(int $bytes, int $intSize): JsonBinaryDecoderValue
+    private function getOffsetOrInLinedValue(int $bytes, int $intSize, int $valueEntrySize): JsonBinaryDecoderValue
     {
         $type = $this->binaryDataReader->readUInt8();
+
         if (self::isInLinedType($type, $intSize)) {
-            return $this->parseScalar($type);
+            $scalar = $this->parseScalar($type);
+
+            // In binlog format, JSON arrays are fixed width elements, even though type value can be smaller.
+            // In order to properly process this case, we need to move cursor to the next element, which is on position 1 + $valueEntrySize (1 is length of type)
+            if($type === self::UINT16 || $type === self::INT16) {
```

Comment:
https://github.com/go-mysql-org/go-mysql/blob/master/replication/json_binary.go#L207 
go implementation is not consuming data, but instead calculates offsets. 
Tbh I dont like this code much, but I have no idea how to make it better (suggestions are welcome).